### ISSUE TITLE
[Navigation Menu] [Accessibility] [Firefox-specific] Keyboard focus not landing on first element in bar

### ISSUE THREAD

**Issue by ellevargas** [radix-ui/primitives#3396]
### Current Behavior

I've got a NavBar component that follows the pattern laid out in the Navigation Menu primitive; the component has a Home link, a Login trigger, and a Contact trigger. When a keyboard user tabs into the space, Login is highlighted first even though it is the second item in the bar. Home can be tabbed back to, but it doesn't receive the initial focus the way it should.

This is only happening in Firefox. Chrome appears unaffected.

### Expected behavior

A keyboard user tabbing into the space should see the first link in the NavBar - Home - focused first.

### Reproducible example

https://codesandbox.io/p/sandbox/navigation-67p4p8

### Additional context

![Image](https://github.com/user-attachments/assets/6bfa9eec-47b9-4273-913e-ab63a7752fb4)

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) | radix-ui/react-navigation-menu | 1.2.5 |
| React            | n/a     | 19.0.0 |
| Browser          | Firefox | 135.0.1 |
| Assistive tech   | n/a        |         |
| Node             | n/a     | 23.2.0 |
| npm/yarn         | npm | 10.9.0 |
| Operating System | MacOS Sequoia | 15.3.1 |
